doc/SHOW_FORMATS_OPTION.md

```diff
@@ -0,0 +1,229 @@
+# SHOW FORMATS OPTION
+
+This file tries to document `--show=formats` option (and a bit of deprecated `--show=types`).
+
+john loads hashes of one format from given hash files. john gives hints about some other formats seen in the file. `--show=formats` will try every line against every format (with a few exceptions) and show formats that can load line independently for each line. It does not mean that john can load all lines at the same time (in one session).
+
+The exceptions:
+- some dynamic formats are disabled in configuration file, they may be enabled temporarily specifying the format with `--format=` option with exact name,
+- `*-ztex` formats are not tried during `--show=formats`, they cannot be enabled (it may be changed in future versions).
+
+Basic uses include:
+- manual and automatic investigation of a file with hashes, hash type identification,
+- easier understanding of cases when john loads only a part of file, but `--show=invalid` may be more convenient to use,
+- help in automated matching of canonical hashes in .pot files with hashes in original form.
+
+## Output and manual investigation
+
+john with `--show=formats` will parse specified hash file and print information about each line in JSON format. JSON format is simple and almost readable.
+
+Example: 2 descrypt hashes:
+```
+$ cat ab.pw
+AAa6CzJlsalyo
+BBODHXVAdtcmc
+```
+
+(`--format=descrypt` is specified to reduce output.)
+```
+$ ./JohnTheRipper/run/john --show=formats ab.pw --format=descrypt
+[{"lineNo":1,"ciphertext":"AAa6CzJlsalyo","rowFormats":[{"label":"descrypt","prepareEqCiphertext":true,"canonHash":["AAa6CzJlsalyo"]}]},
+{"lineNo":2,"ciphertext":"BBODHXVAdtcmc","rowFormats":[{"label":"descrypt","prepareEqCiphertext":true,"canonHash":["BBODHXVAdtcmc"]}]}]```
+
+Example: 1 descrypt hash and pretty printing:
```

Comment:
At least the GitHub ML viewer misses the ``` on line 31 because there's no NL before it. Then the rest of file show as "reversed", all pre-formatted is plain and vice versa

Reply:
There is no NL actually. I did not try to render. My bad.